models/db.py

```diff
@@ -89,6 +89,26 @@
   migrate='runestone_courses.table'
 )
 
+
+# Provide a common query. Pass ``db.courses.ALL`` to retrieve all fields; otherwise, only the ``course_name`` and ``base_course`` are selected.
+def get_course_row(*args, **kwargs):
+    if not args:
+        args = db.courses.course_name, db.courses.base_course
+    return db(db.courses.id == auth.user.course_id).select(*args).first()
+
+
+# Provide the correct URL to a book, based on if it's statically or dynamically served. This function return URL(*args) and provides the correct controller/function based on the type of the current course (static vs dynamic).
+def get_course_url(*args):
+    # Redirect to old-style statically-served books if it exists; otherwise, use the dynamically-served controller.
+    if os.path.exists(os.path.join(request.folder, 'static', auth.user.course_name)):
```

Comment:
In the older style of tests, you have to explicitly setup all the properties of requests.... We had no reason to setup  requests.folder prior to this

Reply:
That what I expected. Another reason to port these tests as well...

Can you work on this? I'm busy until next Tuesday with grading/turning in final grades.

Reply:
Yes, I fixed that and the syntax error and tests are now passing except for a mysterious

```
Ran 7 tests in 9.051s

OK
Traceback (most recent call last):
  File "/srv/web2py/gluon/shell.py", line 282, in run
    execfile(startfile, _env)
  File "applications/runestone/tests/test_assignments.py", line 264, in <module>
    sys.exit(0)
SystemExit: 0
```
At the end of each test.   I don't remember that showing up before, so I'm not sure why it started now.

Reply:
Great, thanks! Fantastic.

I agree -- this is odd. I don't remember seeing that before either.

Reply:
I wish there was some way I could update your PR

Reply:
Is there anything I can do on my side? Give you permission to my repo, I suppose?